src/analysis/retail/monk/windwalker/modules/apl/AplCheck.tsx

```diff
@@ -1,169 +1,135 @@
+import {
+  ASCENSION_ENERGY_MAX_ADDITION,
+  BASE_ENERGY_MAX,
+  INNER_PEACE_ENERGY_MAX_ADDITION,
+} from '../resources/EnergyCapTracker';
 
-const inSerenity = buffPresent(TALENTS.SERENITY_TALENT);
-inSerenity.describe = (tense) => '';
-const notSerenity = not(inSerenity);
-notSerenity.describe = (tense) => '';
-const andSerenity = (cond: Condition<any>) => {
-  const andCnd = and(inSerenity, cond);
-  andCnd.describe = cond.describe;
-  return andCnd;
-};
-const andNotSerenity = (cond: Condition<any>) => {
-  const andCnd = and(not(inSerenity), cond);
-  andCnd.describe = cond.describe;
-  return andCnd;
-};
-
-const needsFaelineHarmony = and(
-  hasTalent(TALENTS.JADEFIRE_HARMONY_TALENT),
-  buffMissing(SPELLS.FAELINE_HARMONY_BUFF),
-);
-
-inSerenity.describe = (tense) => '';
 const hasChi = (min: number) => hasResource(RESOURCE_TYPES.CHI, { atLeast: min });
 
-export const serenityApl = build(
-  [
-    {
-      spell: TALENTS.JADEFIRE_STOMP_TALENT,
-      condition: needsFaelineHarmony,
-    },
-    TALENTS.STRIKE_OF_THE_WINDLORD_TALENT,
-    TALENTS.FISTS_OF_FURY_TALENT,
-    TALENTS.RISING_SUN_KICK_TALENT,
-    SPELLS.BLACKOUT_KICK,
-    {
-      spell: SPELLS.SPINNING_CRANE_KICK,
-      condition: buffPresent(SPELLS.DANCE_OF_CHI_JI_BUFF),
-    },
-    TALENTS.CHI_WAVE_TALENT,
-  ].map((rule: Rule) => {
-    if ('condition' in rule) {
-      return { spell: rule.spell, condition: andSerenity(rule.condition) };
-    }
-    if ('description' in rule) {
-      return { ...rule, condition: inSerenity };
-    }
-    return { spell: rule, condition: inSerenity };
-  }),
-);
-
-export const nonSerenityApl = build(
-  [
-    {
-      spell: TALENTS.JADEFIRE_STOMP_TALENT,
-      condition: needsFaelineHarmony,
-    },
-    {
-      spell: TALENTS.STRIKE_OF_THE_WINDLORD_TALENT,
-      condition: hasChi(2),
-    },
-    TALENTS.WHIRLING_DRAGON_PUNCH_TALENT,
-    {
-      spell: TALENTS.RISING_SUN_KICK_TALENT,
-      condition: and(buffPresent(SPELLS.PRESSURE_POINT_BUFF), hasChi(2)),
-    },
-    {
-      spell: TALENTS.FISTS_OF_FURY_TALENT,
-      condition: hasChi(3),
-    },
-    {
-      spell: TALENTS.RISING_SUN_KICK_TALENT,
-      condition: hasChi(2),
-    },
-    {
-      spell: SPELLS.SPINNING_CRANE_KICK,
-      condition: buffPresent(SPELLS.DANCE_OF_CHI_JI_BUFF),
-    },
-    {
-      spell: SPELLS.BLACKOUT_KICK,
-      condition: describe(and(hasChi(1), not(lastSpellCast(SPELLS.BLACKOUT_KICK))), (tense) => (
+export const apl = build([
+  {
+    spell: SPELLS.TIGER_PALM,
+    condition: describe(
+      and(
+        hasResource(RESOURCE_TYPES.CHI, { atMost: 4 }),
+        hasResource(RESOURCE_TYPES.ENERGY, {
+          // over 90% of max energy
+          atLeast:
+            (BASE_ENERGY_MAX +
+              (hasTalent(TALENTS.ASCENSION_TALENT) ? ASCENSION_ENERGY_MAX_ADDITION : 0) +
+              (hasTalent(TALENTS.INNER_PEACE_TALENT) ? INNER_PEACE_ENERGY_MAX_ADDITION : 0)) *
+            0.9,
```

Comment:
I'd make this .8 I think? Back of the napkin math tells me above that if you spend a global on something else you'll cap energy

Reply:
I ended up splitting the diff and going .85 since I don't want it to be too aggressive, but also I'm hoping to update this percentage with a will cap resource check, I just didn't dig in on whether that was possible